Makefile

```diff
@@ -37,7 +38,7 @@ release-snapshot: ## run a release
 .PHONY: release-snapshot
 
 build: ## build the binary
-	go build ${LDFLAGS} .
+	go build ${LDFLAGS} -o $(BINARY_NAME) .
```

Comment:
I don't think this build will be at BINARY_NAME, since it is not built by goreleaser

Reply:
Can you explain this a bit more when you get the chance? I modified the goreleaser files to refer to the same binaries:
```
builds:
  - binary: terraform-provider-snowflake_{{ .Version }}
```

Reply:
Do the binaries themselves have to be generated from the goreleaser files?

Reply:
binaries get build 2 different ways–

`make build` -> `go build .` -> './terraform-provider-snowflake'

`make release-*` -> `goreleaser ...` -> 'dist/something/terraform-provider-snowflake_version' (with this change)

To see for yourself, you can run `git clean -fdx`, which will remove all files not tracked by git (things that are .gitignore'd) and then run just one of make build|release and see the results.

Reply:
Hmm, so when I ran `make build`, I generated terraform-provider-snowflake_<version>. I haven't been running make release, but `make release-prerelease` executes `make build` first

Reply:
Oh i think i slightly misunderstood what was going on here. This change is probably fine then.